src/libs/actions/Session/index.js

```diff
@@ -250,6 +249,10 @@ function signIn(password, twoFactorAuthCode) {
             createTemporaryLogin(authToken, email);
         })
         .catch((error) => {
+            if (error.message === 'passwordForm.error.twoFactorAuthenticationEnabled') {
```

Comment:
I don't get this. Where is `error.message` coming from? How could it equal `'passwordForm.error.twoFactorAuthenticationEnabled'??
Also, we should not base logic on error messages, we should use the jsonCodes.

Reply:
[Here](https://github.com/Expensify/App/blob/08c2b09bac88bc9e0b43625f5af9631efecabcd3/src/libs/API.js#L261) is where we're throw the error if we receive a 402 json code when authenticating. Maybe it would be better to return the response in API.js [here](https://github.com/Expensify/App/blob/08c2b09bac88bc9e0b43625f5af9631efecabcd3/src/libs/API.js#L261) if the json code is 402 and then change the logic here to:

```suggestion
            if (response.jsonCode === 402) {
```

Reply:
I think we use both as in [here](https://github.com/Expensify/App/blob/6d0baa32e0f1655f8a3b27b192efb2cf626dc1fc/src/libs/actions/ReimbursementAccount/setupWithdrawalAccount.jsjsonCode%20===%20402#L89)
We should check for the json code first, and then if the specific mesage is important then check against it as well, the message should be a constant though, I think

Reply:
@sketchydroide that link is broken, can you re-share it?

Reply:
I believe this is the correct link? https://github.com/Expensify/App/blob/6d0baa32e0f1655f8a3b27b192efb2cf626dc1fc/src/libs/actions/ReimbursementAccount/setupWithdrawalAccount.js#L89

Reply:
Oh I see. Kind of odd we do that... I guess since it is repeating the pattern this is ok.

Reply:
that was some weird fuckery on the link, I've fixed it